### PR TITLE
Use Paging.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Use the [AndroidX Paging](https://developer.android.com/jetpack/androidx/releases/paging) library.

#### Fixes the following issue(s)

- Fixes #

#### Relies on the following changes

- https://github.com/TeamNewPipe/NewPipe/pull/8683

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by TacoTheDank** [comment]
This PR contains the commits from the Lifecycle PR. Could you fix this please?

File: app/build.gradle [hunk 1]
```diff
@@ -218,6 +219,8 @@ dependencies {
     implementation 'androidx.viewpager2:viewpager2:1.1.0-beta01'
     implementation "androidx.work:work-runtime-ktx:${androidxWorkVersion}"
     implementation "androidx.work:work-rxjava3:${androidxWorkVersion}"
+    implementation "androidx.paging:paging-runtime:${androidxPagingVersion}"
```
Comment (TacoTheDank) [inline app/build.gradle:222]: Please keep the AndroidX libraries in alphabetical order (applies to the ext block too).

File: app/src/main/java/org/schabi/newpipe/fragments/list/search/SuggestionListPagingAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        fun onSuggestionItemLongClick(item: SuggestionItem)
+    }
+
+    override fun onBindViewHolder(holder: SuggestionItemHolder, position: Int) {
+        TODO("Not yet implemented")
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SuggestionItemHolder {
+        TODO("Not yet implemented")
+    }
```
Comment (TacoTheDank) [inline app/src/main/java/org/schabi/newpipe/fragments/list/search/SuggestionListPagingAdapter.kt:24]: TODO detected, is there something to be done here?

**Review by TacoTheDank — COMMENTED** [review]
I like the idea of this a lot.

Some comments:

**Comment by Stypox** [comment]
Interesting! Since I don't know much about good practices for Android design, if you could provide some insight into how you did things I would be really glad :-)

**Comment by Isira-Seneviratne** [comment]
It seems like making use of the pagination library will require changes to the extractor as well, since it assumes that the additional page items will be of the same type as the initial ones.

**Comment by Stypox** [comment]
I think each and every list present in NewPipe should be the same, and should hence be able to display all kinds of info items. E.g. info items that come to my mind are: video, channel, playlist, local playlist, video from history, subscription, ... So `Paging` should be made so that it can handle all item types we use at the same time. Does this help you?

**Comment by Isira-Seneviratne** [comment]
> I think each and every list present in NewPipe should be the same, and should hence be able to display all kinds of info items. E.g. info items that come to my mind are: video, channel, playlist, local playlist, video from history, subscription, ... So `Paging` should be made so that it can handle all item types we use at the same time. Does this help you? 

Yeah, definitely. I think it would be best to start with using a view model for the search fragment (I made a separate PR for that) before switching to Paging.

**Pull request closed without merging**